llama-index-integrations/embeddings/llama-index-embeddings-nvidia/llama_index/embeddings/nvidia/base.py

```diff
@@ -183,6 +188,26 @@ def _validate_url(self, base_url):
             warnings.warn(f"{expected_format} Rest is ignored")
         return base_url.strip("/")
 
+    def _validate_model(self, model_name: str) -> None:
+        """
+        Validates compatibility of the hosted model with the client.
+
+        Args:
+            model_name (str): The name of the model.
+
+        Raises:
+            ValueError: If the model is incompatible with the client.
+        """
+        if self._is_hosted:
+            if model_name not in MODEL_ENDPOINT_MAP:
```

Comment:
need to check for model_name in available_models and warn if present, error if not